CIP-0095/README.md

```diff
@@ -0,0 +1,645 @@
+> outlined within
+> [CIP-1694?](https://github.com/cardano-foundation/CIPs/pull/380).
+
+<details>
+  <summary><strong>Wallets and Tooling Hackathon</strong></summary>
+
+  On 2023.07.14 a online and in person community hackathon took place, aims of this event included maturation of the design of this specification.
+
+  We would like to thank the following attendees for providing their valuable insights:
+  - Piotr Czeglik - Lace
+  - Mircea Hasegan - Lace
+  - Alex Apeldoorn - Lace
+  - Michal Szorad - Yoroi
+  - Javier Bueno - Yoroi
+  - Vladimir Volek - Five Binaries
+  - Marek Mahut - Five Binaries
+  - Markus Gufler - Cardano Foundation
+  - Michal Ciborowski - BinarApps
+
+</details>
+
+## Motivation: why is this CIP necessary?
+
+CIP-1694 introduces many new concepts, entities and actors to Cardano;
+describing their implementation at the ledger level. Yet, for most ecosystem
+participants low level details are abstracted away by tools, such as wallets.
+This creates a need for tooling to be able support the utilization of ledger
+features. This specification allows for creation of web-based tools for the
+utilization of CIP-1694's governance features.
+
+This proposal enables Ada holders and DReps to engage web-based tooling through
+wallets. Thus the primary stakeholders for this proposal are tool developers and
+wallet providers. Here we aim to outline all endpoints needed to be exposed to
+web based tools to support all the needs Ada holders and DReps to engage with
+CIP-1694's governance design.
+
+## Specification
+
+We define the following section as an extension to the specification described
+within CIP-30.
+
+To access these functionalities a client application must present this CIP
+extension object:
+
+```ts
+{ "cip": 95 }
+```
+
+This extension object is provided during the initial handshake procedure as
+described within
+[CIP-30's Initial API](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0030#cardanowalletnameenable-extensions-extension----promiseapi).
+
+> **Note** This specification will evolve as the proposed ledger governance
+> model matures.
+
+### DRep Key
+
+// TODO: Move into a separate CIP.
+
+CIP-1694 does not define a derivation path for registered DRep credentials, here
+we propose the introduction of DRep Keys to act as DRep credentials for
+registered DReps.
+
+The methods described here should not be considered the only definitive method
+of generating DRep credentials. Rather these methods should be employed to
+derive keys for non-script registered DReps.
+
+#### Derivation
+
+Here we describe DRep Key derivation as it attains to Cardano wallets who follow
+the
+[CIP-1852 | HD (Hierarchy for Deterministic) Wallets for Cardano](https://github.com/cardano-foundation/CIPs/blob/master/CIP-1852/README.md)
+standard.
+
+To differentiate DRep credentials from other Cardano keys the derivation path
+must follow:
+
+`m / 1718' / 1815' / account' / chain / address_index`
```

Comment:
I would favour `m / 1694' / 1815' / 0' / 1718 / 0` :smiley:
The hardened path is already voting related and cleared for hw-wallets too.

@Ryun1 its like payment and staking keys. totally different, but they are using the same hardened path, only difference is the chain index 0 vs. 2. So i think its would be ok to also reuse existing hardened paths and alter the chain index for such a purpose.

Reply:
I'm seconding @gitmachtl suggestion. I'm not sure why we deviated from 1852' in the first place, but if we've already established precedence that voting keys are in 1694', we might as well run with it. account index should be hardened though, e.g. `/ 1694' / 1815' / 0' / 1718 / 0`

Reply:
As far as I know, any change to the Ledger code base now require an audit and Trezor does not do any audits on Cardano on any change.

Reply:
@gitmachtl @AndrewWestberg @disassembler 

Thank you for your suggestions, I will make the change to:
`m / 1694' / 1815' / 0' / 1718 / 0`

The `1718'` in the purpose field change was inspired by how [CIP-36 vote keys are derived](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0036#derivation-path) (used in Catalyst), but I agree that reusing the purpose of `1694'` from that makes sense.

Reply:
@gitmachtl @AndrewWestberg @disassembler

What do you think of the derivation path:

`m / 1852' / 1815' / account' / 3 / index`

And move this definition into [CIP-1852 | HD Wallets](https://github.com/cardano-foundation/CIPs/blob/master/CIP-1852/README.md)? This would move this key to sit as a _first class_ credential next to Staking Key, which to me seems appropriate.

Reply:
I'm ok with that too. The only thing I am again is brand new hardened path elements which trigger required audits for hardware wallets.

Reply:
@Ryun1 i think staying with the 1852'/1815' path would be ok. people should not mix there normal and there dRep wallet in first place imo. as @disassembler also loves to see it integrated in the 1852'/1815' path i think we should take that route. no new path for the hw wallets, which also means no conflicts maybe with other chains.